src/posts/2020-06-20-tech-stack-matters.md

```diff
@@ -0,0 +1,150 @@
+
+Having way more parts to a tech stack that what you really need to validate your
+product in the market, is something I see time and time again in the wild. It
+usually boils down to two things:
+
+1. Developer overexcitement
+2. Mimicking large companies
+
+Don't get me wrong, developers being excited about things is not bad, but we
+tend to get overexcited with the shiny new toy. It's very common to see huge
+adoption of new technologies, hundreds of blog posts about how they moving from
+technology X to Y was so awesome, only to never hear about it again in a few
+months, while those teams now have to maintain that codebase as it becomes
+apparent that the grass is not greener anymore.
+
+The bigger issue is when small companies try to do what large companies are
+doing. Please, just stop! The one competitive advantage you have as a small
+company is that you are small, you are agile, can pivot quickly, don't throw
+that away. For me the corolary of this are Single Page Application (SPAs),
+there's no good reason for the vast majority of new products to do it, yet it is
+now extremely prevalent. Think of Conway's Law:
+
+>  Any organization that designs a system (defined broadly) will produce a
+>  design whose structure is a copy of the organization's communication
+>  structure.
+
+For a large company, with many different teams and departments, communication
+across them is hard, therefore they design a system that makes that
+communication hard as well. In this case, you have a Front End application that
+only talks to the Back End via a well defined API (contract) and changes to that
+contracts must be agreed upon by both parties. Moreover, both parties will
+implements their own state management, routing, translations and even data
+validations. Obviously there are ways to make these things less obvious and
+(debatably) less painful, but as a small company, why go there in the first
+place?
+
+Again, I advocate for easy to change here, because remember that's what we are
+prioritising. If you really need the interactivity in the client, which is a
+resonable argument to make, use something like PJAX, Stymulus or LiveView (there
+are likely others), who allow you to handle mostly everything from your server
+and remove the duplication of routing, state management, etc. You do lose the
+ability to have the app work offline, which, let's face it, is a fringe scenario
+anyway and in most (if not all) cases not relevant at this stage.
+
+Every line of code is a potential bug, the ideal system does all you want with
+zero lines of code, therefore you want to minimize the amount of code needed to
+achieve your goals. Not using a Front End framework is a good start, but it can
+go to infrastructure to, why not get rid of Docker, Kubernetes, Kafka,
+Tenserflow and so on, and so on. All of these tools have a time and a place to
+be used, don't get me wrong, I just think it should be top of mind for the great
```

Comment:
```suggestion
be used, don't get me wrong. I just think it should be top of mind for the great
```

Reply:
I don't undertand what "should be the top of mind" is suposed to mean

Reply:
It should be "shouldn't"

Reply:
so, "shouldnt' be *on the of anyone's find*" or something like that? still feels incomplete as it is